ibm/service/directlink/resource_ibm_dl_gateway_action.go

```diff
@@ -0,0 +1,1187 @@
+			},
+			dlDefault_import_route_filter: {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validate.InvokeValidator("ibm_dl_gateway_action", dlDefault_import_route_filter),
+				Description:  "The default directional route filter action that applies to routes that do not match any directional route filters",
+			},
+			dlAsPrepends: {
+				Type:        schema.TypeList,
+				Optional:    true,
+				ForceNew:    false,
+				Description: "List of AS Prepend configuration information",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						dlCreatedAt: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The date and time AS Prepend was created",
+						},
+						dlResourceId: {
+							Type:        schema.TypeString,
+							Optional:    true,
+							ForceNew:    false,
+							Computed:    true,
+							Description: "The unique identifier for this AS Prepend",
+						},
+						dlLength: {
+							Type:         schema.TypeInt,
+							Required:     true,
+							ForceNew:     false,
+							ValidateFunc: validate.InvokeValidator("ibm_dl_gateway_action", dlLength),
+							Description:  "Number of times the ASN to appended to the AS Path",
+						},
+						dlPolicy: {
+							Type:         schema.TypeString,
+							Required:     true,
+							ForceNew:     false,
+							ValidateFunc: validate.InvokeValidator("ibm_dl_gateway_action", dlPolicy),
+							Description:  "Route type this AS Prepend applies to",
+						},
+						dlPrefix: {
+							Type:        schema.TypeString,
+							Optional:    true,
+							ForceNew:    false,
+							Description: "Comma separated list of prefixes this AS Prepend applies to. Maximum of 10 prefixes. If not specified, this AS Prepend applies to all prefixes",
+							Deprecated:  "prefix will be deprecated and support will be removed. Use specific_prefixes instead",
+						},
+						dlSpecificPrefixes: {
+							Type:        schema.TypeList,
+							Description: "Array of prefixes this AS Prepend applies to",
+							Optional:    true,
+							ForceNew:    false,
+							MinItems:    1,
+							MaxItems:    10,
+							Elem:        &schema.Schema{Type: schema.TypeString},
+						},
+						dlUpdatedAt: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The date and time AS Prepend was updated",
+						},
+					},
+				},
+			},
+			dlBfdInterval: {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				ForceNew:     false,
+				Description:  "BFD Interval",
+				ValidateFunc: validate.InvokeValidator("ibm_dl_gateway_action", dlBfdInterval),
+			},
+			dlBfdMultiplier: {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				ForceNew:     false,
+				Description:  "BFD Multiplier",
+				ValidateFunc: validate.InvokeValidator("ibm_dl_gateway_action", dlBfdMultiplier),
+			},
+			dlBfdStatus: {
+				Type:        schema.TypeString,
```

Comment:
Computed attributes should n't have optional

Reply:
Fixed